w11_d1_1_node_express_mongo_review/README.md

```diff
@@ -0,0 +1,313 @@
+
+## Project Structure Setup
+
+We already have Node.js and MongoDB installed, so we'll start from that point. Type these instructions in your Terminal to do basic project setup.
+
+  ```bash
+  $ mkdir QnA  # make a directory for your project and cd into it
+  $ cd QnA
+  $ git init  # turn this directory into a git repo
+  $ touch README.md
+  ```
+
+Now, let's set it up as a Node.js project.
+
+  ```bash
+  $ touch server.js  # create server-side js file
+  $ ls    # check that your dir has README.md and server.js
+  $ npm init   # initialize a node project in this directory. fill in the answers to each question
+  $ ls   # note that your dir now has package.json in it
+  $ cat package.json  #  take a look at the config data that `npm install` put into package.json
+  ```
+
+Let's install express and a few other node modules we know we'll need.
+
+  ```bash
+  $ npm install --save express   # our back end framework
+  $ ls     # note that we now have a node_modules folder - thanks npm!
+  $ cat package.json   # also note that package.json now lists express as a dependency for our app
+  $ npm install --save body-parser   # we use this node module to parse data from requests
+  ```
+
+Finally, we know we want to use MongoDB with mongoose, so let's set up a directory and install the module for that.
+
+  ```bash
+  $ mkdir models
+  $ touch models/question.js  # this store our questions model code
+  $ npm install --save mongoose   # like ActiveRecord, provides convenience methods for accessing our DB
+  ```
+
+## Server File Setup  
+
+```js
+// server.js
+
+// require express and other modules
+var express = require('express'),
+    app = express(),
+    bodyParser = require('body-parser'),
+    mongoose = require('mongoose');
+
+// connect to mongodb
+mongoose.connect(
+  process.env.MONGOLAB_URI ||
+  process.env.MONGOHQ_URL ||
+  'mongodb://localhost/REPLACE WITH YOUR APP DB NAME'
+);
+
+// configure body-parser
+app.use(bodyParser.urlencoded({extended: true}));
+
+
+
+// listen on port 3000
+app.listen(process.env.PORT || 3000, function () {
+  console.log('server started on localhost:3000');
+});
+```
+
+
+
+## Model
+
+```js
+// models/question.js
+
+var mongoose = require('mongoose'),
+    Schema = mongoose.Schema;
+
+var QuestionSchema = new Schema({
+  text: String
+});
+
+var Question = mongoose.model('Question', QuestionSchema);
+
+module.exports = Question;
+```
+
+## Routes
+
+If we think about a Rails analogy, our `server.js`  file for Express apps has been where we write our routes and our controller code.  Let's look at three routes - a GET for all questions, a  POST to create a new question, and a PATCH to update parts of an existing question.
+
+```js
+// server.js
+
+// send back all questions
+app.get('/questions', function (req, res) {
+  Question.find({}, function (err, questions) {
```

Comment:
Before using the `Question` model, maybe add a line about defining `Question` in `server.js` (`var Question = require('./models/question');`)